src/admin/runners/updateCollectionsRunner.js

```diff
@@ -107,12 +117,39 @@ class UpdateCollectionsRunner {
             `Connecting to target cluster with mongo url: ${process.env.SOURCE_CLUSTER_MONGO_URL} and db_name: ${db_name}`
         );
         const options = {
-            retryWrites: true,
-            w: 'majority',
+            // https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/connection-options/
+            connectTimeoutMS: 0,
+            maxIdleTimeMS: 0,
+            serverSelectionTimeoutMS: 600000 // Wait for 60 seconds before server selection is complete.
         };
         return { connection: mongoUrl, db_name: db_name, options: options };
     }
 
+    /**
+     * @description given an array of collection filters out only the required ones.
+     * @param {Array} collectionList
+     * @return {Array}
+     */
+    getListOfCollections(collectionList) {
+        let collectionNames = [];
+        for (const collection of collectionList) {
+            // If the collection of type view, we can skip it
+            if (collection.type !== 'collection') {
+                continue;
+            }
+            // If the list of collection is mentioned verify the collection name is in the list of collections passed
+            if (this.collections && !this.collections.includes(collection.name)) {
+                continue;
+            }
+            // If history collections are to be skipped
+            if (this.skipHistoryCollections && collection.name.endsWith('_History')) {
+                continue;
+            }
+            collectionNames.push(collection.name);
+        }
```

Comment:
We should exclude 'system' collections if not already excluded. Ref https://github.com/icanbwell/fhir-server/blob/6071b179143b88c181bbbedad5fc1ca11f1648f3/src/utils/mongoCollectionManager.js#L69